File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def solution(n, k):
    from collections import deque
    r,t=0,''
    # 진수 변환
    while n>0:
        t=str(n%k)+t
        n//=k
    t=t.split('0')
    for num in t:
        if num!='' and is_prime(int(num)):
            r+=1
    return r

def is_prime(num):
    if num<2:
        return False
    for i in range(2,int(num**0.5)+1):
        if num%i==0:
            return False
    return True

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.cpp
#include <string>
#include <vector>
#include <sstream>
#include <cmath>

using namespace std;

bool is_prime(long num) {
    if(num<2)
        return false;
    for(int i=2;i<(int)pow(num,0.5)+1;i++) {
        if(num%i==0)
            return false;
    }
    return true;
}

int solution(int n, int k) {
    int r=0;
    string t;
    while(n>0) {
        t=to_string(n%k)+t;
        n/=k;
    }
    stringstream ss(t);
    while(getline(ss,t,'0')) {
        if(!t.empty() and is_prime(stol(t)))
            r+=1;
    }
    return r;
}